### PR TITLE
Viewpager implementation on home page

### PR DESCRIPTION
**Pull request by camilo05**
Sep-21-2021 10-32-53

### CODE REVIEW

File: app/src/main/res/values/ic_launcher_background.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <color name="ic_launcher_background">#FFFFFF</color>
```
Comment (rodrigo103) [inline app/src/main/res/values/ic_launcher_background.xml]: Why is this not in the colors.xml file?

File: app/src/main/res/layout/fragment_profile.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:tools="http://schemas.android.com/tools"
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
```
Comment (rodrigo103) [inline app/src/main/res/layout/fragment_profile.xml]: `xmlns:app="http://schemas.android.com/apk/res-auto"` and `xmlns:tools="http://schemas.android.com/tools"` are not used in this layout

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package ar.com.wolox.android.example.ui.home.news
+
+import ar.com.wolox.android.R
+import ar.com.wolox.android.databinding.FragmentNewsBinding
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import javax.inject.Inject
+
+class NewsFragment @Inject constructor() : WolmoFragment<FragmentNewsBinding, NewsPresenter>(), NewsView {
```
Comment (rodrigo103) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt:8]: Is this `@Inject constructor()` needed

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package ar.com.wolox.android.example.ui.home.news
+
+import ar.com.wolox.wolmo.core.presenter.BasePresenter
+import javax.inject.Inject
+
+class NewsPresenter @Inject constructor() : BasePresenter<NewsView>()
```
Comment (rodrigo103) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsPresenter.kt:6]: Is this `@Inject constructor()` needed?

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt (hunk 3 above)
Comment (LisandroDiMeo) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt:8]: I think he will need it on later cards 🤔

**Review by LisandroDiMeo — APPROVED** [review]
Great job !! 👏 🥇

File: app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerView.kt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package ar.com.wolox.android.example.ui.home.viewpager
+
+interface ViewPagerView {
```
Comment (rodrigo103) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerView.kt]: We usually put these inside their corresponding fragment.

File: app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerFragment.kt [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            override fun onPageSelected(position: Int) {
+                presenter.onSelectedTab(position)
+            }
+        })
+        presenter.onSelectedTab(0)
+    }
+
+    override fun setChangeColorIcon(position: Int) {
+        binding.tabLayout.apply {
+            when (position) {
```
Comment (rodrigo103) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerFragment.kt]: Here the view is making a decision. Remember that decisions go in the presenter.

File: app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerFragment.kt [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                    getTabAt(1)?.setIcon(R.drawable.ic_profile_on)
+                }
+            }
+        }
+    }
+
+    companion object {
+        fun newInstance() = ViewPagerFragment()
+    }
+}
```
Comment (rodrigo103) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerFragment.kt:72]: Create an enum class for the tabs and use it in this fragment. Something like this:
```
        enum class Tabs(val id: Int, val tabName: String, val iconOn: Int, val iconOff: Int) {
            NEWS(0, "News", R.drawable.ic_news_list_on, R.drawable.ic_news_list_off),
            PROFILE(1, "Profile", R.drawable.ic_profile_on, R.drawable.ic_profile_off)
        }
```

**Review by rodrigo103 — CHANGES_REQUESTED** [review]
Very well done! I suggest some changes, let me know if you have questions about them.

File: app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerFragment.kt [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+            override fun onPageSelected(position: Int) {
+                presenter.onSelectedTab(position)
+            }
+        })
+        presenter.onSelectedTab(0)
+    }
+
+    override fun setChangeColorIcon(list: List<TabsData>) {
+        list.forEach { binding.tabLayout.getTabAt(it.position)?.setIcon(it.icon) }
```
Comment (ffasitella) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/viewpager/ViewPagerFragment.kt:56]: Nice!

File: app/src/main/res/layout/fragment_viewpager.xml [hunk 9]
```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent">
+
+        <androidx.appcompat.widget.Toolbar
```
Comment (ffasitella) [inline app/src/main/res/layout/fragment_viewpager.xml:9]: The design did not mention (at least not as mandatory) to create a Toolbar, but It's perfect this way.

**Review by ffasitella — APPROVED** [review]
Amazing job! Keep like that! 🚀

**Review by rodrigo103 — APPROVED** [review]

**Pull request merged**